dist/steps/run_tests.sh

```diff
@@ -65,118 +56,61 @@ echo ""
+  echo "#   Testing in $platform  #"
   echo "###########################"
   echo ""
-  unity-editor \
-    -batchmode \
-    -logFile "$FULL_ARTIFACTS_PATH/editmode.log" \
-    -projectPath "$UNITY_PROJECT_PATH" \
-    -runTests \
-    -testPlatform editmode \
-    -testResults "$FULL_ARTIFACTS_PATH/editmode-results.xml" \
-    $CUSTOM_PARAMETERS
-
-  # Catch exit code
-  EDIT_MODE_EXIT_CODE=$?
-
-  # Print unity log output
-  cat "$FULL_ARTIFACTS_PATH/editmode.log"
 
-  # Display results
-  if [ $EDIT_MODE_EXIT_CODE -eq 0 ]; then
-    echo "Run succeeded, no failures occurred";
-  elif [ $EDIT_MODE_EXIT_CODE -eq 2 ]; then
-    echo "Run succeeded, some tests failed";
-  elif [ $EDIT_MODE_EXIT_CODE -eq 3 ]; then
-    echo "Run failure (other failure)";
+  if [[ "$platform" != "COMBINE_RESULTS" ]]; then
+    runTests="-runTests -testPlatform $platform -testResults $FULL_ARTIFACTS_PATH/$platform-results.xml"
   else
-    echo "Unexpected exit code $EDIT_MODE_EXIT_CODE";
+    runTests="-quit"
   fi
-fi
 
-#
-# Testing in PlayMode
-#
-PLAY_MODE_EXIT_CODE=0
-if [ "$PLAY_MODE" = "true" ]; then
-  echo ""
-  echo "###########################"
-  echo "#   Testing in PlayMode   #"
-  echo "###########################"
-  echo ""
   unity-editor \
     -batchmode \
-    -logFile "$FULL_ARTIFACTS_PATH/playmode.log" \
+    -logFile "$FULL_ARTIFACTS_PATH/$platform.log" \
     -projectPath "$UNITY_PROJECT_PATH" \
-    -runTests \
-    -testPlatform playmode \
-    -testResults "$FULL_ARTIFACTS_PATH/playmode-results.xml" \
+    -coverageResultsPath "$FULL_COVERAGE_RESULTS_PATH" \
+    $runTests \
+    -enableCodeCoverage \
+    -debugCodeOptimization \
+    -coverageOptions "$COVERAGE_OPTIONS" \
     $CUSTOM_PARAMETERS
 
   # Catch exit code
-  PLAY_MODE_EXIT_CODE=$?
+  TEST_EXIT_CODE=$?
 
   # Print unity log output
-  cat "$FULL_ARTIFACTS_PATH/playmode.log"
+  cat "$FULL_ARTIFACTS_PATH/$platform.log"
 
   # Display results
-  if [ $PLAY_MODE_EXIT_CODE -eq 0 ]; then
+  if [ $TEST_EXIT_CODE -eq 0 ]; then
     echo "Run succeeded, no failures occurred";
-  elif [ $PLAY_MODE_EXIT_CODE -eq 2 ]; then
+  elif [ $TEST_EXIT_CODE -eq 2 ]; then
     echo "Run succeeded, some tests failed";
-  elif [ $PLAY_MODE_EXIT_CODE -eq 3 ]; then
+  elif [ $TEST_EXIT_CODE -eq 3 ]; then
     echo "Run failure (other failure)";
   else
-    echo "Unexpected exit code $PLAY_MODE_EXIT_CODE";
+    echo "Unexpected exit code $TEST_EXIT_CODE";
   fi
-fi
-
-#
-# Results
-#
-
-echo ""
-echo "###########################"
-echo "#    Project directory    #"
-echo "###########################"
-echo ""
-ls -alh $UNITY_PROJECT_PATH
 
-if [ "$EDIT_MODE" = "true" ]; then
-  echo ""
-  echo "###########################"
-  echo "#    Edit Mode Results    #"
-  echo "###########################"
-  echo ""
-  cat "$FULL_ARTIFACTS_PATH/editmode-results.xml"
-  cat "$FULL_ARTIFACTS_PATH/editmode-results.xml" | grep test-run | grep Passed
-fi
+  if [ $TEST_EXIT_CODE -gt 0 ]; then
```

Comment:
While we're at it, this should be `-ne 0` since failed exit codes can be negative.

Reply:
I didn't know that I'll add the change!